update deprecated fields in kustomization documentation (#48805)
kubernetes/website: content/en/docs/tasks/manage-kubernetes-objects/kustomization.md

#### Customizing

Patches can be used to apply different customizations to Resources. Kustomize supports different patching
mechanisms through `StrategicMerge` and `Json6902` using the `patches` field. `patches` may be a file or 

> an inline string, targetting a single or multiple resources.

Reviewer: Sorry there was a typo in my previous suggestion.

```suggestion
an inline string, targeting a single or multiple resources.
```

> The `patches` field contains a list of patch applied in the order they are specified, the patch target 

Reviewer: I think this should be 2 sentences.

```suggestion
The `patches` field contains a list of patches applied in the order they are specified. The patch target
```

> selects resources by `group`, `version`, `kind`, `name`, `namespace`, `labelSelector` and `annotationSelector`

Reviewer: Missing a period at the end of the sentence.

```suggestion
selects resources by `group`, `version`, `kind`, `name`, `namespace`, `labelSelector` and `annotationSelector`.
```

            memory: 512Mi
```

Not all Resources or fields support `strategicMerge` patches. To support modifying arbitrary fields in arbitrary Resources,
Kustomize offers applying [JSON patch](https://tools.ietf.org/html/rfc6902) through `Json6902`.

> To find the correct Resource for a `Json6902` patch, it manadatory to specific the `target` field in `kustomization.yaml`. 

Reviewer 2: ```suggestion
To find the correct Resource for a `Json6902` patch, it is mandatory to specific the `target` field in `kustomization.yaml`. 
```

> To find the correct Resource for a `Json6902` patch, it is manadatory to specify the `target` field in `kustomization.yaml`. 

Reviewer: ```suggestion
To find the correct Resource for a `Json6902` patch, it is mandatory to specify the `target` field in `kustomization.yaml`. 
```

set of resources and associated customization. A base could be either a local directory or a directory from a remote repo,
as long as a `kustomization.yaml` is present inside. An **overlay** is a directory with a `kustomization.yaml` that refers to other
kustomization directories as its `bases`. A **base** has no knowledge of an overlay and can be used in multiple overlays.

> The `kustomization.yaml` in a **overlay** directory may refer to multiple `bases`, combining all the resources defined 

Reviewer 3: (nit)
This should be
```suggestion
The `kustomization.yaml` in an _overlay directory_ may refer to multiple `bases`, combining all the resources defined
```

set of resources and associated customization. A base could be either a local directory or a directory from a remote repo,
as long as a `kustomization.yaml` is present inside. An **overlay** is a directory with a `kustomization.yaml` that refers to other
kustomization directories as its `bases`. A **base** has no knowledge of an overlay and can be used in multiple overlays.

> An overlay may have multiple bases and it composes all resources from bases and may also have customization on top of them.

Reviewer 2: This sentence is a little awkward, even rephrased. Is it possible to provide more clarity here?

```suggestion
An overlay may have multiple bases, and composes all resources from bases. It may also have customizations on top of them.
```
